### PR TITLE
Fix not showing arrow-function's params as completions

### LINKED ISSUE #33494: Parameter not visible in the body of an arrow function
Consider the following:
```ballerina
function (int) returns int fn3 = (p) => p + <cursor>y;
```
In the function body, the parameter `p` should be visible.

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #33494 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java [hunk 1]
```diff
@@ -605,6 +610,19 @@ public void visit(BLangOnFailClause onFailClause) {
         this.acceptNode(onFailClause.body, this.symbolEnv);
     }
 
+    @Override
+    public void visit(BLangArrowFunction arrowFunctionNode) {
+        SymbolEnv arrowFunctionSymbolEnv = SymbolEnv.createArrowFunctionSymbolEnv(arrowFunctionNode, this.symbolEnv);
+        arrowFunctionNode.params.forEach(param -> symbolEnter.defineNode(param, arrowFunctionSymbolEnv));
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java:616]: We shouldn't be using nor do we need to use `SymbolEnter` here. This means you are modifying the AST and potentially altering semantics.

**Comment by mohanvive** [comment]
@dulajdilshan can you please check the review comments and fix the conflict.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java [hunk 2]
```diff
@@ -203,6 +203,12 @@ public static SymbolEnv createResourceActionSymbolEnv(BLangInvokableNode node, S
 
     public static SymbolEnv createArrowFunctionSymbolEnv(BLangArrowFunction node, SymbolEnv env) {
         SymbolEnv symbolEnv = cloneSymbolEnvForClosure(node, env);
+        Scope scope = node.body.scope;
+        if (scope == null) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java:207]: In what sort of scenarios does this become `null`? 🤔

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java:207]: This becomes `null` at the compiler level (when we creating the arrow-func-symbol environemtn

**Review by mohanvive — APPROVED** [review]

**Pull request merged**